frontend/apps/filemanager/filemanager.lua

```diff
@@ -850,66 +831,143 @@ function FileManager:cutFile(file)
+    local text, ok_text
+    if self.cutfile then
+        text = _("Move selected files to the current folder?")
+        ok_text = _("Move")
+    else
+        text = _("Copy selected files to the current folder?")
+        ok_text = _("Copy")
+    end
+    local confirmbox, check_button_overwrite
+    confirmbox = ConfirmBox:new{
+        text = text,
+        ok_text = ok_text,
+        ok_callback = function()
+            close_callback()
+            self:pasteSelectedFiles(check_button_overwrite.checked)
+        end,
+    }
+    check_button_overwrite = CheckButton:new{
+        text = _("overwrite existing files"),
+        checked = true,
+        parent = confirmbox,
+    }
+    confirmbox:addWidget(check_button_overwrite)
+    UIManager:show(confirmbox)
+end
+
+function FileManager:pasteSelectedFiles(overwrite)
+    local dest_path = BaseUtil.realpath(self.file_chooser.path)
+    local ok_files = {}
+    for orig_file in pairs(self.selected_files) do
+        local orig_name = BaseUtil.basename(orig_file)
+        local dest_file = BaseUtil.joinPath(dest_path, orig_name)
+        local ok
+        local dest_mode = lfs.attributes(dest_file, "mode")
+        if not dest_mode or (dest_mode == "file" and overwrite) then
+            if self.cutfile then
+                ok = self:moveFile(orig_file, dest_path)
+            else
+                ok = self:copyRecursive(orig_file, dest_path)
+            end
+        end
         if ok then
+            DocSettings.updateLocation(orig_file, dest_file, not self.cutfile)
+            ok_files[orig_file] = true
+            self.selected_files[orig_file] = nil
+        end
+    end
+    local skipped_nb = util.tableSize(self.selected_files)
+    if util.tableSize(ok_files) > 0 then
+        if self.cutfile then -- for move only
+            ReadHistory:updateItems(ok_files, dest_path)
+            ReadCollection:updateItems(ok_files, dest_path)
+        end
+        if skipped_nb > 0 then
             self:onRefresh()
-            self.clipboard = nil
         end
     end
-
-    local mode = lfs.attributes(dest_file, "mode")
-    if mode then
-        UIManager:show(ConfirmBox:new{
-            text = mode == "file" and T(_("File already exists:\n%1\nOverwrite file?"), BD.filename(orig_name))
-                                   or T(_("Folder already exists:\n%1\nOverwrite folder?"), BD.directory(orig_name)),
-            ok_text = _("Overwrite"),
-            ok_callback = function()
-                doPaste()
-            end,
+    if skipped_nb > 0 then -- keep select mode on
+        local text1, text2
+        if self.cutfile then
+            text1 = "1 file was not moved"
+            text2 = "%1 files were not moved"
+        else
+            text1 = "1 file was not copied"
+            text2 = "%1 files were not copied"
+        end
+        UIManager:show(InfoMessage:new{
+            text = T(N_(text1, text2, skipped_nb), skipped_nb),
```

Comment:
@Frenzie : will this be properly picked up by the translatable strings extraction stuff?

Reply:
It won't, it's just a relatively dumb text match.